vm/vm.go

```diff
@@ -74,10 +75,12 @@ func (vm *VM) Run(program *Program, env any) (_ any, err error) {
 	} else {
 		vm.Stack = vm.Stack[0:0]
 	}
-
 	if vm.Scopes != nil {
 		vm.Scopes = vm.Scopes[0:0]
 	}
+	if vm.Variables == nil {
+		vm.Variables = make([]any, program.variables)
```

Comment:
nevermind if you're still working on it, but since I'm on the topic - should the VM be reusable for different programs? 

because then i think we need to resize the variables slice here depending on the number of variables in the current program, something like

```go
if cap(vm.Variables) < len(program.variables) { 
    // we need to allocate more in this case
    vm.Variables = make([]any, len(program.variables)) 
} else {
    // we can resize the already-allocated slice
    vm.Variables = vm.Variables[:len(program.variables)] 
}
```

Reply:
Correct suggestion. Usually it make sense to only run a vm for a specific program (as we this way stack will be reallocated and potently will not grow)